crates/librqbit/src/storage/examples/inmemory.rs

```diff
@@ -108,14 +144,29 @@ impl TorrentStorage for InMemoryExampleStorage {
             lengths: self.lengths,
             map: RwLock::new(map),
             file_infos: self.file_infos.clone(),
+            max_ram_size_per_torrent: self.max_ram_size_per_torrent,
         }))
     }
 
-    fn init(&mut self, _meta: &crate::ManagedTorrentInfo) -> anyhow::Result<()> {
+    fn init(&mut self, _meta: &crate::ManagedTorrentShared) -> anyhow::Result<()> {
         Ok(())
     }
 
     fn remove_directory_if_empty(&self, _path: &Path) -> anyhow::Result<()> {
         Ok(())
     }
+
+    fn on_piece_completed(&self, file_id: usize, offset: u64) -> anyhow::Result<()> {
```

Comment:
it would make more sense to pass ValidatedPieceIndex to it rather than file_id and offset.
Piece can span multiple files, so file_id and offset are kind of meaningless.

Reply:
Done in 9183df0ebd14b3026028cda077b0757128facaed. This indeed makes more sense